app/lib/primer/classify/functional_colors.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Primer
+  class Classify
+    # https://primer-css-git-mkt-color-modes-docs-primer.vercel.app/css/support/v16-migration
+    class FunctionalColors
+      FUNCTIONAL_TEXT_OPTIONS = {
+        primary: :text_primary,
+        secondary: :text_secondary,
+        tertiary: :text_tertiary,
+        link: :text_link,
+        success: :text_success,
+        warning: :text_warning,
+        danger: :text_danger,
+        white: :text_white,
+        inverse: :text_inverse
+      }.freeze
+
+      TEXT_COLOR_MAPPINGS = {
+        gray_dark: FUNCTIONAL_TEXT_OPTIONS[:primary],
+        gray: FUNCTIONAL_TEXT_OPTIONS[:secondary],
+        gray_light: FUNCTIONAL_TEXT_OPTIONS[:tertiary],
+        blue: FUNCTIONAL_TEXT_OPTIONS[:link],
+        green: FUNCTIONAL_TEXT_OPTIONS[:success],
+        yellow: FUNCTIONAL_TEXT_OPTIONS[:warning],
+        red: FUNCTIONAL_TEXT_OPTIONS[:danger],
+        white: FUNCTIONAL_TEXT_OPTIONS[:white],
+        # still unsure what will happen with these colors
+        black: nil,
+        orange: nil,
+        orange_light: nil,
+        purple: nil,
+        pink: nil,
+        inherit: nil
```

Comment:
Currently Primer CSS still offers the [`.text-inherit`](https://github.com/primer/css/blob/474f27fca4ffd9cbf78e057923648e0b90f3101a/src/utilities/colors.scss#L51-L52). I was debating about removing it, but I think it's still useful and would be good to keep it.

Something we could consider is renaming it from `.text-inherit` to `.color-text-inherit`. But it's not really a color that comes from `primer/primitives` but rather just a way to "reset" a color. So maybe fine to keep as is.

Reply:
I believe that is the correct behavior, using the presentational class when a functional one doesn't exist.